tools/worker-runner/tc/fakeworkermanager.go

```diff
@@ -50,13 +51,39 @@ func (wm *FakeWorkerManager) RemoveWorker(workerPoolID, workerGroup, workerID st
 	return nil
 }
 
+func (wm *FakeWorkerManager) ReportWorkerError(workerPoolID string, payload *tcworkermanager.WorkerErrorReport) (*tcworkermanager.WorkerPoolError, error) {
+	workerPoolError := tcworkermanager.WorkerPoolError{
+		Description:  payload.Description,
+		Extra:        payload.Extra,
+		ErrorID:      "1",
+		Kind:         payload.Kind,
+		Reported:     tcclient.Time(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC)),
+		WorkerPoolID: workerPoolID,
+	}
+	wmWorkerErrorReports = append(wmWorkerErrorReports, payload)
+	return &workerPoolError, nil
+}
+
+func FakeWorkerManagerWorkerErrorReports() ([]*tcworkermanager.WorkerErrorReport, error) {
+	if len(wmWorkerErrorReports) == 0 {
+		return nil, fmt.Errorf("No reportWorkerError calls")
+	} else {
+		reports := make([]*tcworkermanager.WorkerErrorReport, len(wmWorkerErrorReports))
+		copy(reports, wmWorkerErrorReports)
+		wmWorkerErrorReports = []*tcworkermanager.WorkerErrorReport{}
+		return reports, nil
+	}
+}
+
 // Get the single registration that has occurred, or an error if there are not
 // exactly one.  This resets the list of registrations in the process.
 func FakeWorkerManagerRegistration() (*tcworkermanager.RegisterWorkerRequest, error) {
 	if len(wmRegistrations) == 0 {
 		return nil, fmt.Errorf("No registerWorker calls")
 	} else if len(wmRegistrations) == 1 {
-		return wmRegistrations[0], nil
+		req := wmRegistrations[0]
+		wmRegistrations = []*tcworkermanager.RegisterWorkerRequest{}
```

Comment:
`wmRegistrations` wasn't getting reset between tests, although the comment above notes that it is intended to be.

Reply:
Sorry!